### PR TITLE
fix(core): synchronize state of open from `tuiDropdownHover`

### LINKED ISSUE #3372: 🐞 - `TuiHostedDropdown` doesn't reflect open state when `tuiDropdownHover` is used
### Which @taiga-ui/* package(s) are the source of the bug?

core

### Please provide a link to a minimal reproduction of the bug

https://stackblitz.com/edit/angular-argxkc?file=src%2Fapp%2Fapp.component.html

### Is this issue blocking you?

Non-Blocking

### Description

The open state of the `TuiHostedDropdown` is always its initial value.

### Angular version

15.0.0

### Taiga UI version

3.15.0

### Which browsers have you used?

- [X] Chrome
- [ ] Firefox
- [ ] Safari
- [X] Edge

### Which operating systems have you used?

- [ ] macOS
- [X] Windows
- [ ] Linux
- [ ] iOS
- [ ] Android

### PR DESCRIPTION
**Pull request by splincode**
## PR Checklist

Please check if your PR fulfills the following requirements:

- [x] The commit message follows [Conventional Commits](https://www.conventionalcommits.org/en/)
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

## PR Type

What kind of change does this PR introduce?

- [x] Bugfix
- [ ] Feature
- [ ] Refactoring
- [ ] Code style update
- [ ] Build or CI related changes
- [ ] Documentation content changes

## What is the current behavior?

Closes #3372

## What is the new behavior?

image

## Does this PR introduce a breaking change?

- [ ] Yes
- [ ] No

### CODE REVIEW

File: projects/core/components/hosted-dropdown/hosted-dropdown.component.ts [hunk 1]
```diff
@@ -94,6 +96,11 @@ export class TuiHostedDropdownComponent implements TuiFocusableElementAccessor {
         @Inject(TuiDropdownHoverDirective)
         private readonly hover$: Observable<boolean> | null,
         @Inject(ElementRef) private readonly elementRef: ElementRef,
+        @Optional()
+        @Self()
+        @Inject(TuiDropdownHoverDirective)
+        readonly dropdownHover: TuiDropdownHoverDirective | null,
+        @Inject(TUI_IS_CYPRESS) readonly isCypress: boolean,
```
Comment (vladimirpotekhin) [inline projects/core/components/hosted-dropdown/hosted-dropdown.component.ts:103]: I don't think components should know about cypress

File: projects/core/directives/dropdown/dropdown-hover.directive.ts [hunk 2]
```diff
@@ -12,10 +12,16 @@ export class TuiDropdownHoverDirective extends TuiDriver {
     private readonly toggle$ = new Subject<boolean>();
     private readonly stream$ = merge(this.toggle$, this.hovered$).pipe(
         switchMap(visible =>
-            of(visible).pipe(delay(visible ? this.showDelay : this.hideDelay)),
+            of(visible).pipe(
+                delay(visible ? this.showDelay : this.hideDelay),
+                tap(visible => this.openChange.emit(visible)),
```
Comment (vladimirpotekhin) [inline projects/core/directives/dropdown/dropdown-hover.directive.ts:17]: this breaks hoverable/clickable case: once clicked, the dropdown shouldn't close until click outside dropdown

https://user-images.githubusercontent.com/46284632/212109236-7063ede3-7956-42b1-9e80-7920cb5f69a2.mov

Comment (splincode) [reply projects/core/directives/dropdown/dropdown-hover.directive.ts:43]: fixed

File: projects/core/components/hosted-dropdown/hosted-dropdown.component.ts (hunk 1 above)
Comment (splincode) [reply projects/core/components/hosted-dropdown/hosted-dropdown.component.ts]: fixed

**Pull request closed without merging**